gcp/app-engine/db_utils.py

```diff
@@ -0,0 +1,158 @@
+    db_name = os.getenv('DB_NAME', default='postgres')
+    db_instance_conn_name = os.getenv('INSTANCE_CONNECTION_NAME', default='stoked-depth-428423-j7:us-central1:golf-better')
+    db_host = os.getenv('DB_HOST', default=f'/cloudsql/{db_instance_conn_name}')
+    db_port = os.getenv('DB_PORT', default=5432)
+
+    pw_log = "****" if db_password is not None else "<unset>"
+    print(f'Connecting to PG on user={db_user}, pw={pw_log}, host={db_host}, port={db_port}, db={db_name}')
+
+    def getconn():
+        if db_instance_conn_name:
+            print(f'Connecting to CloudSQL instance with instance name: "{db_instance_conn_name}"')
+            connector = Connector()
+            return connector.connect(
+                db_instance_conn_name,
+                "pg8000",
+                user=db_user,
+                password=db_password,
+                db=db_name,
+                #ip_type=IPTypes.PRIVATE
+            )
+        else:
+            print('Connecting to vanilla Postgres database')
+            return pg8000.connect(
+                user=db_user,
+                password=db_password,
+                host=db_host,
+                port=db_port,
+                database=db_name
+            )
+
+    pool = sqlalchemy.create_engine(
+        "postgresql+pg8000://",
+        creator=getconn,
+        connect_args={
+            "port": db_port
+        }
+    )
+
+    return pool
+
+def get_next_tournaments(tournament_search=None):
+    query = f"""
+        select *
+        from
+        (
+          select name,
+            start_date,
+            is_completed,
+            data->>'date' as date,
+            data->>'city' as city,
+            coalesce(data->>'stateCode', '') as stateCode,
+            data->>'countryCode' as countryCode
+          from {DB_SCHEMA}.pga_tournaments
+          where ({f"lower(id) like '%{str(tournament_search).lower()}%' or lower(name) like '%{str(tournament_search).lower()}%'" if tournament_search is not None else "1=1"})
+          order by is_completed, abs(EXTRACT(EPOCH FROM (NOW() - start_date)))
+          limit {1 if tournament_search is not None else 5}
+        )
+        order by start_date
+    """
+    pool = get_db_connection()
+    now = datetime.now()
+    with pool.connect() as db_conn:
+        records = db_conn.execute(sqlalchemy.text(query)).fetchall()
+        tournaments = [{
+            "name": row[0],
+            "location": row[4] + (", " + row[5] + ' ' if row[5] else ' '),
+            "date": row[3],
+            "status": "Done" if row[2] else "Scheduled" if now.date() < row[1] else "In Progress"
+        } for row in records]
+        if len(tournaments) == 1:
+            return tournaments[0]
+        return tournaments
+
+def get_player(player_search, tournament_search, round=None):
+    tournament_filter = f"lower(t.id) like '%{str(tournament_search).lower()}%' or lower(t.name) like '%{str(tournament_search).lower()}%'"
```

Comment:
i made all of these files do a case-insensitive substring, so you can do something like "@Golf Better API player info thomps wyndh" to get player info about Thompson Davis at the Wyndham. I still need to clean up this quite a bit since it doesn't support spaces in the commands, etc., but is at least a start